lib/pure/collections/sets.nim

```diff
@@ -241,8 +241,11 @@ iterator items*[A](s: HashSet[A]): A =
   ##   assert a.len == 2
   ##   echo b
   ##   # --> {(a: 1, b: 3), (a: 0, b: 4)}
+  let length = s.len
   for h in 0 .. high(s.data):
-    if isFilled(s.data[h].hcode): yield s.data[h].key
+    if isFilled(s.data[h].hcode):
+      yield s.data[h].key
+      assert(len(s) == length, "the length of the HashSet changed while iterating over it")
```

Comment:
how about use an auxiliary template:
```nim
template checkLen(a, b) =
  const s = "the length of " & $type(a) & " changed while iterating over it"
  assert len(a) == length, s
```
makes it easy to change in 1 place (eg if we want to add runtime info `$(len(a), length)` for extended debugging)